frontend/web/src/app/components/physical-print/physical-print-details-panel/physical-print-details-panel.component.ts

```diff
@@ -1,18 +1,91 @@
-import {Component, Input, OnInit} from '@angular/core';
+import {Component, EventEmitter, Input, OnChanges, OnInit, Output} from '@angular/core';
+import {PhysicalPart} from '../../../model/physical-part';
 import {PhysicalPrint} from '../../../model/physical-print';
-
+import {PhysicalPartService} from '../../../services/physical-part/physical-part.service';
+declare let $: any;
 
 @Component({
   selector: 'app-physical-print-details-panel',
   templateUrl: './physical-print-details-panel.component.html',
   styleUrls: ['./physical-print-details-panel.component.scss'],
+  providers: [PhysicalPartService],
 })
-export class PhysicalPrintDetailsPanelComponent implements OnInit {
+export class PhysicalPrintDetailsPanelComponent implements OnInit, OnChanges {
+  private table;
+  private dataLoaded = false;
+  private loadingTable = false;
+
   @Input('physicalPrint') physicalPrint: PhysicalPrint = null;
+  @Input('physicalParts') physicalParts: PhysicalPart [] = null;
+  @Output() closeReq: EventEmitter<null> = new EventEmitter<null>();
 
-  constructor() { }
+  constructor(private physicalPartService: PhysicalPartService) {
+    /* get parts */
+    this.physicalPartService.getPhysicalParts().subscribe((pParts) => {
+      this.physicalParts = pParts;
+      this.dataLoaded = true;
+      this.loadTable();
+    });
+  }
 
   ngOnInit() {
+    this.table = $('#pPartsTable');
   }
 
+  ngOnChanges() {
+    if (this.dataLoaded && !this.loadingTable && this.table) { // make sure we truly have loaded the parts
+      this.loadTable();
+    }
+  }
+
+
+  private loadTable(){
+    this.loadingTable = true;
+    this.populate();
+    this.loadingTable = false;
+  }
+
+  private populate() {
+    const _self = this;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "_self". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1854)